### PR TITLE
feat: add wab widgets post

### PR DESCRIPTION
**Pull request by stdavis**
Pull request description
------------------------

### CODE REVIEW

**Comment by stdavis** [comment]
@atribolet What do you think about the commit that I just pushed? It adds more of a conclusion/summary.

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+### 3.Use Polyfill.io.
+Using babel worked great for this project until I needed polyfills for older browsers. (I'm looking at you, Internet Explorer.) Loading third-party libraries via AMD from within WAB widgets [is a pain](https://developers.arcgis.com/web-appbuilder/sample-code/add-a-third-party-library.htm). Fortunately, I found [Polyfill.io](https://polyfill.io/v3/). This is a service that dynamically returns a set of polyfills as defined by query parameters. The cool part about this service is that it only loads the code required for the browser that is making the request. Using this library from within WAB requires [adding it to the `loadResource` method in `init.js`](https://github.com/agrc/wfrc-wab-widgets#add-a-polyfill-for-older-browsers).
+
+Through working on this project, I feel like I've got a much better understanding of WAB and how it fits into our offerings. There is definite potential as long as it's shortcomings are well understood by the developer and client. I hope that these tips save you some headaches if you decide that WAB is a good fit for your next project. {% capture contact %}{% include contact.html
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: For the first line here, just a couple of minor surface edits: 

Through working on this project, I feel like I now have a much better understanding of WAB and how it fits into our offerings at AGRC.

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* a best practice folder structure;
+* grunt tasks for JS tooling, including babel, livereload, and sass; and
+* a grunt task to automate the copying of your development code to WAB developer edition.
+
+Esri also includes a generator for scaffolding out the widgets themselves. This project provided a great starting point for my work.
+
+### 2. Write your WAB widgets as dojo widgets.
+Another great tip that I found was to write your WAB widgets as plain old dojo widgets (inherit from `dijit._WidgetBase`) and then wrap them for WAB. This made unit tests much easier. The other benefit to this approach is that it makes it much easier to reuse the widgets in non-WAB applications. Here is an example repo: [gbochenek/wab-test-example](https://github.com/gbochenek/wab-test-example).
+
+### 3.Use Polyfill.io.
```
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: Small thing: add a space here between 3. and Use for consistency with the 1. and 2. ;)

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md (hunk 1 above)
Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: For the second sentence, again, some super minor changes: 

There is definite potential with WAB as long as its shortcomings are well understood by the developer and the client.

Comment (atribolet) [inline _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: I think this closing sums everything up really nicely, @stdavis :)

Comment (stdavis) [reply _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: ✅

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md (hunk 2 above)
Comment (stdavis) [reply _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: ✅

File: _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md (hunk 1 above)
Comment (stdavis) [reply _posts/2019-01-25-best-practices-for-building-web-appbuilder-widgets.md]: ✅

**Comment by steveoh** [comment]
👍

**Pull request merged**